ThomazFB/WordPress-Android
# Replace UI and internal Giphy references

Summary
------------
This Pull Request brings renaming for all Giphy references inside package, classes, documentation and UI strings. Giving sequence to the [issue 11456](https://github.com/wordpress-mobile/WordPress-Android/issues/11456).

Here's the change log:

* Every UI string with `Giphy` information was replaced with `Tenor`. 

* The `giphy` packages inside `ui` and `viewmodel` were both renamed to only `gif`.

* All classes with the `Giphy` naming convention were renamed to only `Gif` to better fit the agnostic gif provider implementation strategy.

* All comments and documentation with references to `Giphy` or the Giphy SDK were readapted to better fit the `GifProvider` structure.

* Removed the Giphy SDK  reference at maven repository in `build.gradle`

Testing
------------

### Before testing ###
1. Create a Tenor API key and paste it associated with the `wp.tenor.api_key` as the value inside your `gradle.properties`
2. When testing verify that all UI strings correctly reference Tenor as the gif provider.

### Testing from the Post Editor ###
1. Click to create a new post
2. Click on the overflow button at the top right end of the screen
3. select the `Switch to classic editor` 
4. Click on the plus (+) button at the bottom left end of the screen
5. Select the `Choose from Tenor` option
6. Type in the search bar to query GIFs
7. Once inside the Gif Picker, make sure that `selection`, `preview` and `add` are all working as expected

### Testing from the Media Browser ###
1. Go to the Media Browser
2. Click on the Add button at the top right. Select the `Choose from Tenor` option
3. Type in the search bar to query GIFs
4. Once inside the Gif Picker, make sure that `selection`, `preview` and `add` are all working as expected


PR submission checklist
------------
